scipy/interpolate/_rbfinterp.py

```diff
@@ -387,6 +389,73 @@ def __init__(self, y, d,
         self.epsilon = epsilon
         self.powers = powers
 
+    def _chunk_evaluator(
+            self,
+            x,
+            y,
+            shift,
+            scale,
+            coeffs,
+            memory_budget=None
+    ):
+        """
+        Evaluate the interpolation why controlling memory consumption.
+        We chunk the input if we need more memory than specified.
+
+        Parameters
+        ----------
+        x : (Q, N) float ndarray
+            array of points on which to evaluate
+        y: (P, N) float ndarray
+            array of points on which we know function values
+        shift: (N, ) array
+            Domain shift used to create the polynomial matrix.
+        scale : (N,) float ndarray
+            Domain scaling used to create the polynomial matrix.
+        coeffs: (P+R, S) float ndarray
+            Coefficients in front of basis functions
+        memory_budget: int
+            Total amount of memory (in units of sizeof(float)) we wish
+            to devote for storing the array of coefficients for
+            interpolated points. If we need more memory than that, we
+            chunk the input.
+
+        Returns
+        -------
+        (Q, S) float ndarray
+        Interpolated array
+        """
+        nx, ndim = x.shape
+        if self.neighbors is None:
+            nnei = len(y)
+        else:
+            nnei = self.neighbors
+        # in each chunk we consume the same space we already occupy
+        chunksize = memory_budget // ((self.powers.shape[0] + nnei)) + 1
```

Comment:
The default `memory_budget` is None?

Reply:
This was the decision to let the caller to always make the decision on the memory_budget. I.e. the default memory budget is set in the top level functions based on the total size of arrays. I'm sure it can be done differently, but it made sense to me.

Reply:
It's fine in principle, the issue is that None should likely be replaced by a numerical value

Reply:
I can put the default value to 1e6 if you think it's better. The disadvantage from my poiint of view is that the 1e6 value will be in now two locations in the code.

Reply:
okay, I've put the default to 1e6